config/config.go

```diff
@@ -165,6 +167,32 @@ func (cfg *AuctionTimeouts) validate(errs []error) []error {
 	return errs
 }
 
+type Compression struct {
+	Request  CompressionConfig `mapstructure:"request"`
+	Response CompressionConfig `mapstructure:"response"`
+}
+
+// CompressionConfig defines if compression is enabled and what type of compression to use
+type CompressionConfig struct {
+	Enabled bool   `mapstructure:"enabled"`
+	CType   string `mapstructure:"type"`
```

Comment:
Could you call this field `Type`? If not due to language restrictions, consider `Kind`.

Reply:
Yes, it was due to language restrictions. Will rename to `Kind`